### ISSUE TITLE
Measurable test coverage: Silver requirement?

### ISSUE THREAD

**Issue by WilcoFiers** [w3c/wcag3#387]
A challange in WCAG 1 and 2 is that it's impossible to say anything about conformance unless a web page is fully tested. Because you don't always have a complete picture, it is difficult to say anything about the accessibility. It'd be very helpful if in addition to allowing pages to conform if not 100% of the content passes the requirements, to also allow conformance claims of pages that aren't 100% tested.

This is going to be relevant when testing websites too. How much of the site should be tested before someone can claim conformance? Currently I think this is only ever done by counting numbers of pages. But I don't think that has to be the only way to do it.

This could in theory work if applicability of a requirement could be made programatically determinable. For instance, if a requirement is applicable to all img elements, and a page has 100 of them, with 90 img elements tested, you have 90% test coverage for that requirements. WCAG 2 doesn't have this concent of applicability, making a measurement like this impossible.

If we allow incidental issues in conformance of Silver, it makes sense to also allow test coverage to be lower than 100%. By writing Silver in such a way that it can be used to measure test coverage, it becomes possible to maintain conformance over time. Rather than have the conformance claim get outdated the moment a page changes, conformance claim can stay relevant as long as the test target didn't change too much. Even better, you can programatically determine the moment the conformance claim becomes outdated, as that will be the moment the test coverage drops below a certain level.

**Comment by TestPartners** [comment]
My background was in functional testing before I moved into accessibility, and I can assure you that there is not and cannot be a statistically valid way to measure test coverage, let alone measure it accurately or meaningfully, so we must not go down that path.

One of the many problems is that of fungibility. To take the example of a page containing 100 images, those images are not equivalent in any way, so counting them is meaningless. Some will be decorative, some will be non-decorative but low importance, and some will be non-decorative and essential. Some images may be in dialogs, accordions, carousels or other types of content that is not initially displayed, so they will be viewed far less often than other images. Each image will affect the accessibility to a different extent.

Testers do report test coverage metrics, but they are all bogus. Some charlatan test consultants have made an entire career of it. The topic has been discussed for at least four decades in the functional testing community, so if there was a solution it would have been found by now. But there isn't and there won't be.

If you insist that test coverage measurements must be made, you must at least add a statement that they don't mean anything.

**Comment by alastc** [comment]
Overall, we are somewhat limited in what we can do within conformance. Due to the issues @TestPartners mentioned, including a recommended / set coverage is not possible if we want it to be take up by regulators etc.

However, we are starting work on separate guidance documents (policy / testing / reporting), and I think this idea would be useful input to at least the testing guidance, if not all of them.

I've updated the label from Conformance to Testing Guidance, please comment if you think that is not appropriate.
